Volume configuration in Kraftfile causes a kraftkit crash / invalid memory address / nil pointer dereference

### Describe the bug

Adding a volume in the Kraftfile brings kraftkit to a crash upon executing run.

(Adding the -v flag does not work. Does not result in crash, but then haproxy will complain about missing haproxy.conf (meaning it did not get passed through)

### Steps to reproduce

Adding:
```
volumes:
- ./haproxy:/etc/haproxy/
```
to the Kraftfile of the haproxy configuration (from the catalog)

### Expected behavior

No crashing :)

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
> kraft run --log-type basic --log-level debug --rm -p 8081:80 -p 8404:8404 --plat qemu --arch x86_64 -M 256M


 D  kraftkit 0.9.2
 D  using platform=qemu
 D  determining how to proceed given provided input and context
 D  using compatible context candidate=kraftfile-unikraft
 D  found volume strategy volume strategy=9pfs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x120 pc=0x1ba098d]

goroutine 1 [running]:
kraftkit.sh/internal/cli/kraft/run.(*RunOptions).parseKraftfileVolumes(0xc005fca310?, {0x257e9f8, 0xc0004eb740}, {0x25a3260, 0xc006440360}, 0xc0003b4588)
        /__w/kraftkit/kraftkit/internal/cli/kraft/run/utils.go:323 +0x68d
kraftkit.sh/internal/cli/kraft/run.(*runnerKraftfileUnikraft).Prepare(0xc000624f00, {0x257e9f8, 0xc0004eb740}, 0xc0001a2b60, 0xc0003b4588, {0x0?, 0xf?, 0xc0006bb550?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/run/runner_kraftfile_unikraft.go:201 +0xb8b
kraftkit.sh/internal/cli/kraft/run.(*RunOptions).Run(0xc0001a2b60, {0x257e9f8, 0xc0004eb740}, {0xc00025af00, 0x0, 0xf})
        /__w/kraftkit/kraftkit/internal/cli/kraft/run/run.go:335 +0xe5a
kraftkit.sh/cmdfactory.New.func1(0xc0000fc608?, {0xc00025af00?, 0x0?, 0x0?})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:372 +0x43
kraftkit.sh/cmdfactory.AttributeFlags.bind.func3(0xc0000fc608, {0xc00025af00, 0x0, 0xf})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:554 +0x1cb
github.com/spf13/cobra.(*Command).execute(0xc0000fc608, {0xc00025ae10, 0xf, 0xf})
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc0002c1508)
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1034
kraftkit.sh/cmdfactory.Main({0x257e9f8, 0xc0004eacf0}, 0xc0002c1508)
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:184 +0x6f
kraftkit.sh/internal/cli/kraft.Main({0x7fff0b5d6444?, 0x35f0398?, 0xc0000061c0?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/kraft.go:207 +0xa46
main.main()
        /__w/kraftkit/kraftkit/cmd/kraft/main.go:19 +0x79
```

--- a/internal/cli/kraft/run/utils.go
+++ b/internal/cli/kraft/run/utils.go
@@ -320,7 +320,7 @@ func (opts *RunOptions) parseKraftfileVolumes(ctx context.Context, project app.A
 			},
 		})
 
-		if err == nil && vol.Spec.Source != "" {
+		if err == nil && vol != nil && vol.Spec.Source != "" {
 			vol.Spec.Destination = volcfg.Destination()
 			machine.Spec.Volumes = append(machine.Spec.Volumes, *vol)
 			continue

